features/eolearn/features/feature_manipulation.py

```diff
@@ -252,38 +252,46 @@ class SpatialResizeTask(EOTask):
         resize_method: ResizeMethod = ResizeMethod.LINEAR,
         resize_library: ResizeLib = ResizeLib.PIL,
     ):
         """
         :param features: Which features to resize. Supports new names for features.
-        :param resize_parameters: Instructions on how to perform the resizing process. For example use:
-            `['new_size', [500, 1000]]` to resize the data to size (500, 1000),
-            `['resolution', [10, 20]]` for changing resolution from 10 to 20,
-            `['scale_factors', [3, 3]]` to make the data three times larger.
+        :param resize_type: Determines type of resizing process and how `width_param` and `height_param` are used.
+            Options:
+                * `new_size`: Resizes data to size (width_param, height_param)
+                * | `resolution`: Resizes the data to have width_param, height_param resolution over width/height axis.
+                  | Uses EOPatch bbox to compute.
+                * | `scale_factor` Resizes the data by scaling the width and height by a factor set by
+                  | width_param and height_param respectively.
+        :param height_param: Parameter to be applied to the height in combination with the resize_type
+        :param width_param: Parameter to be applied to the width in combination with the resize_type
         :param resize_method: Interpolation method used for resizing.
         :param resize_library: Which Python library to use for resizing. Default is PIL, as it supports all dtypes and
             features anti-aliasing. For cases where execution speed is crucial one can use CV2.
         """
         self.features = features
-        self.parameter_kind = ResizeParam(resize_parameters[0])
-        self.parameter_values = resize_parameters[1]
+        self.resize_type = ResizeParam(resize_type)
+        self.height_param = height_param
+        self.width_param = width_param
 
         self.resize_function = partial(
             spatially_resize_image, resize_method=resize_method, resize_library=resize_library
         )
 
     def execute(self, eopatch: EOPatch) -> EOPatch:
         resize_fun_kwargs: Dict[str, Any]
-        if self.parameter_kind == ResizeParam.RESOLUTION:
+        if self.resize_type == ResizeParam.RESOLUTION:
             if not eopatch.bbox:
                 raise ValueError("Resolution-specified resizing can only be done on EOPatches with a defined BBox.")
-            new_size = bbox_to_dimensions(eopatch.bbox, self.parameter_values)
-            resize_fun_kwargs = {ResizeParam.NEW_SIZE.value: new_size}
+            new_width, new_height = bbox_to_dimensions(eopatch.bbox, (self.width_param, self.height_param))
+            resize_fun_kwargs = {ResizeParam.NEW_SIZE.value: (new_height, new_width)}
         else:
-            resize_fun_kwargs = {self.parameter_kind.value: self.parameter_values}
+            resize_fun_kwargs = {self.resize_type.value: (self.height_param, self.width_param)}
```

Comment:
Just a comment. I find it confusing that the resolution type is handled here separately, while the case for scaling factor is handled inside of the resize function. 

Additionally, the resize function could be improved, since it's a bunch of optional parameters due to the fact that it offers flexibility.

I would suggest that the logic that was implemented here is moved to the resize function, which would accept these new parameters, and that the function is just use here.

Reply:
resolution cannot be handled inside the functions because it works on numpy arrays, but we need a BBox for the resolution. We could however extract that functionality from the helper function into the task.

Reply:
You're right about the bbox thing, thanks.

Is the function used elsewhere? Perhaps your suggestion would make senes.